packages/postcss-reduce-initial/src/script/lib/io.mjs

```diff
@@ -0,0 +1,37 @@
+import fetch from 'node-fetch';
+import { reduceInitial } from './mdnCssProps.mjs';
+
+export function validate(data) {
+  if (
+    data === undefined ||
+    !Object.keys(data.fromInitial).length ||
+    !Object.keys(data.toInitial).length
+  ) {
+    return Promise.reject(new Error('Fetching & processing JSON data failed!'));
+  }
+  return data;
+}
+
+export function handleError(error) {
+  if (error) {
+    throw error;
+  }
+}
+
+export function toJSONString(data) {
+  return `${JSON.stringify(data, null, 2)}\n`;
+}
+
+export function write(fileFunc, filePaths, data, key) {
```

Comment:
I think this function can be misleading since in practice `key`, `filepaths` and `fileFunc` are always the same. I would get rid of it. The exact names are more important than the mechanism since they're the interface with the plugin code.

Reply:
`fileFunc` is injected to [aid in testability](https://github.com/cssnano/cssnano/pull/1160/files#r666925656), and I chose the structure with `filePaths` + `key` and returning the `data` in part because it allowed cleanly chaining the two file writes in the promise chain.

I felt it was cleaner if `write` only concerned itself with the task of writing a single file. And that the responsibility of directing which and how many files are created belongs more in `generate`. Alternatively that control could be moved to `acquire.mjs` - but that would obscure it from testing.

Another factor leading to the choice of passing in `filePaths` from `acquire.mjs` is that use of `__dirname` to get the base path is no longer allowed by Node in ES modules. So it was necessary to build the paths using the new `import.meta.url` - which didn't play well with `jest`. So this separation simplifies things in that regard.

The chaining and `filePaths` + `key` solution - instead of for example having `write` do both writes - also allowed for writing cleaner tests as the responsibilities were more clearly separated. The test against `generate` asserting that two files are written. The test against `write` making a single assertion to verify that it formats the data before it is passed to `writeFile`. And the `test.each()` also indirectly verifying that `write` does indeed write the specific files we target with the `key` - to make it less fragile.

I'm not sure if I understand what you mean with `"interface with the plugin code"` - could you elaborate?

Reply:
> I'm not sure if I understand what you mean with "interface with the plugin code" - could you elaborate?
I mean the cssnano plugin only sees the file names, all the rest is an implementation detail. In that sense the download script 'interface' are the JSON file names.